### PR TITLE
Upgrade Synapse to 1.19.1

### PR DESCRIPTION
**Pull request by ulope**
## Description

This upgrades the Synapse requirement to 1.18.0 and fixes all backwards incompatibilities that have been introduced.
This is mainly related to the [alias semantics change (MSC2432)](matrix-org/matrix-doc#2432).

Fixes: raiden-network/raiden-service-bundle#180
Fixes: #6424

Due to this being backwards incompatible server side we need to come up with a migration strategy before merging this: #6443 [done]

### CODE REVIEW

File: raiden/network/transport/matrix/client.py
Comment (karlb) [inline raiden/network/transport/matrix/client.py:97]: Something is wrong or hard to read in first part of this sentence.

File: raiden/tests/utils/transport.py
Comment (karlb) [inline raiden/tests/utils/transport.py:161]: Do we really need all these cases, now?

File: raiden/tests/utils/transport.py
Comment (karlb) [inline raiden/tests/utils/transport.py:200]: Did synapse change away from twisted, or does twisted work with the async syntax, now?

File: raiden/utils/cli.py
Comment (karlb) [inline raiden/utils/cli.py:448]: We could also just return all and let the caller choose by indexing the returned dict.

**Review by karlb — APPROVED** [review]
Looks good, I just left some random comments as usual.

File: raiden/tests/utils/transport.py
Comment (ulope) [reply raiden/tests/utils/transport.py:161]: This now covers both cases, the one used in the tests as well as the RSB setup.
The plan is to have this code only once and not two slightly different versions in here and in the RSB.

File: raiden/tests/utils/transport.py
Comment (ulope) [reply raiden/tests/utils/transport.py:200]: They are slowly converting to asyncio. 
This API has already been changed.

I'm not sure what they're using as a compatibility layer, but one of the twisted maintainers works for them so I assume they have that covered.

**Comment by ulope** [comment]
BTW the scenario player test can't succeed until the RSB PR is merged, which needs this PR merged.
I just love circular dependencies...

File: raiden/utils/cli.py
Comment (karlb) [reply raiden/utils/cli.py:448]: It's not. I just like to thing aloud when reviewing PRs.

**Pull request merged**
